jenkins/Jenkinsfile

```diff
@@ -1,374 +1,496 @@
-    string(name: "openidc_proxy", defaultValue: "2.3.1_2",
-          description:"The tag of the openidc proxy image to pull in docker image build. Used in the prepare script.")
-  }
+    options {
+        // We don't want to be building more than one PR's changes at once, we will queue them instead
+        disableConcurrentBuilds()
+    }
 
-  stages {
-    stage('Terra docker image scan') {
-      steps {
-        git credentialsId: 'jenkins-ssh-github', url: 'git@github.com:DataBiosphere/terra-docker.git', branch: 'master'
+    triggers {
+        // Leaving the string empty causes it to trigger off pushes/PRs to the specified branch
+        pollSCM("")
+    }
 
-        script {
-          def temp = sh( 
-            script: "cat config/conf.json | jq -r '.image_data | .[] | select(.automated_flags.include_in_custom_dataproc == true) | select(.automated_flags.build == true) | .name'",
-            returnStdout: true
-          ).trim().split("\n")
+    parameters {
+        booleanParam(name: "GENERATE_DATAPROC_IMAGE", defaultValue: true)
+        booleanParam(name: "GENERATE_GCE_IMAGE", defaultValue: true)
+        string(name: "IMAGE_BASE_NAME", defaultValue: "custom-leo-image",
+            description: "The name to give the custom dataproc image. This will be appended with a hyphen and the short version of the commit hash (first 7 characters).")
+        booleanParam(name: "USE_CUSTOM_IMAGE_IDENTIFIER", defaultValue: false)
+        string(name: "CUSTOM_IMAGE_ID", defaultValue: "sampleUniqueID",
+            description: "Specify along with USE_CUSTOM_IMAGE_IDENTIFIER to build the VM image with a specific name")
+        string(name: "GOOGLE_PROJECT", defaultValue: "broad-dsp-gcr-public",
+            description: "The google project to use")
+        string(name: "DATAPROC_IMAGE_BUCKET", defaultValue: "gs://leo-dataproc-image",
+            description: "The path for the bucket where the custom Dataproc images will be stored")
+        string(name: "GCE_IMAGE_BUCKET", defaultValue: "gs://leo-gce-image",
+            description: "The path for the bucket where the custom GCE images will be stored")
+        string(name: "DATAPROC_VERSIONS", defaultValue: "1.2.79-debian9 1.4.15-debian9",
+            description: "A custom image will be build for each of these dataproc versions")
+        booleanParam(name: "USE_CUSTOM_DOCKER_IMAGE_VERSIONS", defaultValue: false,
+            description: "Check if you want to specify custom image versions below")
+        string(name: "terra_jupyter_base", defaultValue: "0.0.6")
+        string(name: "terra_jupyter_hail", defaultValue: "0.0.5")
+        string(name: "terra_jupyter_r", defaultValue: "0.0.7")
+        string(name: "terra_jupyter_python", defaultValue: "0.0.6")
+        string(name: "terra_jupyter_bioconductor", defaultValue: "0.0.9")
+        string(name: "terra_jupyter_gatk", defaultValue: "0.0.8")
```

Comment:
Also https://github.com/DataBiosphere/leonardo/pull/1307 adds an AoU image here so that will have to be merged :)

Reply:
Btw I know these are just defaults but they're a bit out of date -- `terra-jupyter-gatk` for example is up to `0.0.13`. Can we make the defaults more recent?

Reply:
certainly 👍